python/packages/autogen-core/src/autogen_core/components/models/_openai_client.py

```diff
@@ -393,18 +403,32 @@ async def create(
 
         if len(tools) > 0:
             converted_tools = convert_tools(tools)
-            future = asyncio.ensure_future(
-                self._client.chat.completions.create(
-                    messages=oai_messages,
-                    stream=False,
-                    tools=converted_tools,
-                    **create_args,
+            if use_beta_client:
+                future = asyncio.ensure_future(
+                    self._client.beta.chat.completions.parse(
+                        messages=oai_messages,
+                        tools=converted_tools,
+                        **create_args,
```

Comment:
`response_format` needs to be passed in explicitly here to the result type specified.

Reply:
i have done this, but i have other type checking error. will need more time to fix this